core/src/main/java/org/apache/accumulo/core/data/Condition.java

```diff
@@ -308,41 +309,20 @@ public Condition setIterators(IteratorSetting... iterators) {
 
   @Override
   public boolean equals(Object o) {
-    if (o == this) {
+    if (this == o)
       return true;
-    }
-    if (o == null || !(o instanceof Condition)) {
-      return false;
-    }
-    Condition c = (Condition) o;
-    if (!(c.cf.equals(cf))) {
-      return false;
-    }
-    if (!(c.cq.equals(cq))) {
-      return false;
-    }
-    if (!(c.cv.equals(cv))) {
+    if (o == null || getClass() != o.getClass())
```

Comment:
This was changed from `instanceof` to the use of `getClass()`, which is not the same. The latter is more narrow. I think it should be changed back to `instanceof`, although the check for null would be redundant, since `instanceof` is always false if it's null.

Reply:
@ctubbsii, I reverted the aforementioned check back to its original form. Thanks.